app/uk/gov/hmrc/trackingconsentfrontend/assets/test/renderBanner.spec.ts

```diff
@@ -1,13 +1,46 @@
+/* global spyOn */
+
 import {
   queryByText
 } from '@testing-library/dom'
 import '@testing-library/jest-dom/extend-expect'
+import userPreferenceFactory from '../src/userPreferenceFactory'
 import renderBanner from '../src/ui/renderBanner'
 
 describe('index', () => {
+  const reset = () => {
+    document.getElementsByTagName('html')[0].innerHTML = ''
+  }
+  const click = (elem) => {
+    const event = document.createEvent('HTMLEvents')
+    event.initEvent('click', false, true)
+    elem.dispatchEvent(event)
+  }
+  const assume = expect
+
+  beforeAll(reset)
+  afterEach(reset)
+
   it('should render a banner', () => {
-    renderBanner()
+    renderBanner(userPreferenceFactory())
```

Comment:
Given we're using a factory pattern, would it make sense to use it and pass in a stub rather than having to use the real implementation and spy on it? Otherwise what value is this indirection giving us?

Reply:
That's fair.  I'll update.